kernel/work.c

```diff
@@ -646,6 +646,8 @@ static void work_queue_main(void *workq_ptr, void *p2, void *p3)
 			 * submissions.
 			 */
 			(void)z_sched_wake_all(&queue->drainq, 1, NULL);
+		} else {
```

Comment:
The top commit at https://github.com/pabigot/zephyr/commits/pr/33198 provides a more helpful description, if you'd like to incorporate it.

Reply:
Thank you! I will cherry-pick to include this improvement :)